### PR TITLE
debian: Run as ndn user with its home directory

### PR DESCRIPTION
**Pull request by pwithnall**
Create and run as an `ndn` user with HOME=/var/lib/ndn, the same as in
the ndn-cxx package. Keys will be installed to /var/lib/ndn/.ndn by
programs from both packages, and the key store will be shared.

As part of this, the nfd binary needs to have some additional
capabilities set, which adds a build-dep on libcap2-bin for setcap.

We can’t run the nfd daemon as the ndn user from the start (using
systemd’s User key) as it does its own socket setup, which requires root
privileges. The nfd configuration has been changed to drop privileges
and run as ndn after startup. nfd does not support systemd (or inetd)
socket activation.

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T13932

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -31,7 +31,8 @@ Vcs-Browser: https://github.com/named-data/NFD
 
 Package: nfd
 Architecture: any
-Depends: lsb-base (>= 3.0-6),
+Depends: libcap2-bin,
```
Comment (xaiki) [inline debian/control:34]: shouldn't that be in another commit ?

File: debian/nfd.postinst [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+if [ "$1" = configure ]; then
+  # Add a user for all NDN related files and processes
+  adduser \
+      --system \
+      --quiet \
+      --home /var/lib/ndn \
+      --disabled-password \
+      --group \
+      ndn
```
Comment (xaiki) [inline debian/nfd.postinst]: if we're adding this in libndn *and* NFD bad things will happen.

File: nfd.conf.sample.in [hunk 3]
```diff
@@ -5,8 +5,8 @@ general
   ; when not performing privileged tasks. NFD does not drop
   ; privileges by default.
 
-  ; user ndn-user
-  ; group ndn-user
+  user ndn
+  group ndn
```
Comment (xaiki) [inline nfd.conf.sample.in]: as on the other commit, ndn is a very generic name.

**Review by xaiki — COMMENTED** [review]
- [ ] supperfluous libpcap addition
- [ ] ndn is not a good name
- [ ] we shouldn't create the same user in ndn-cxx and nfd, as nfd depends on ndn-cxx it's probably best to do the creation of the user (and the keys) there.

File: debian/nfd.postinst (hunk 2 above)
Comment (pwithnall) [reply debian/nfd.postinst]: From https://wiki.debian.org/AccountHandlingInMaintainerScripts:
> It should normally not be necessary to cross-check with getent whether an account already exists since adduser --system generally does the right thing

File: debian/control (hunk 1 above)
Comment (pwithnall) [reply debian/control:34]: No, it’s needed for `setcap`, which is added in this commit.

**Comment by pwithnall** [comment]
> we shouldn't create the same user in ndn-cxx and nfd, as nfd depends on ndn-cxx it's probably best to do the creation of the user (and the keys) there.

It should be fine to have the `adduser` call in both packages, as it is a no-op if the user already exists. Both packages directly depend on that user existing, so it should be created in both of them — if we set it up so only `libndn-cxx0.5.0` created the user, and `nfd` depended on that happening, things would break if we later removed the `postinst` script from `libndn-cxx0.5.0` (if the user was no longer needed there, for example).

**Comment by xaiki** [comment]
ok, i don't like to repeat code... but well...

**Comment by pwithnall** [comment]
Updated to address review comments (renaming the user).
